### PR TITLE
Added exception for invalid dtypes per issue #15520

### LINKED ISSUE #15520: BUG: invalid dtypes should raise
xref https://github.com/dask/dask/issues/1965

when an explict dtype is provided, we should raise if its not a valid ``numpy/pandas`` dtype.

```
In [1]: pd.Series([], name='time',dtype=pd.Timestamp)
Out[1]: Series([], Name: time, dtype: object)
```

### PR DESCRIPTION
**Pull request by trevorpburke**
close #15520 

 - [ ] closes #xxxx
 - [ ] tests added / passed
 - [ ] passes ``git diff upstream/master | flake8 --diff``
 - [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
tests!

File: pandas/core/series.py [hunk 1]
```diff
@@ -156,7 +156,11 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
             if data is None:
                 data = {}
             if dtype is not None:
-                dtype = self._validate_dtype(dtype)
+                if dtype is numpy.dtype or dtype is pandas.dtype:
+                    dtype = self._validate_dtype(dtype)
+                else:
```
Comment (jreback) [inline pandas/core/series.py]: you would change this in the _validate_dtype method

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/series.py (hunk 1 above)
Comment (trevorpburke) [inline pandas/core/series.py]: oh! pandas/pandas/indexes/range.py, right?

Comment (jreback) [inline pandas/core/series.py]: this is in generic.py

**Comment by jreback** [comment]
tests!

**Comment by jreback** [comment]
if you'd like to add some tests, ping and can reopen

**Pull request closed without merging**